src/Sylius/Bundle/ReviewBundle/Doctrine/ORM/Subscriber/LoadMetadataSubscriber.php

```diff
@@ -106,12 +104,12 @@ private function createReviewerMapping($reviewerEntity, ClassMetadata $reviewerE
         return [
             'fieldName' => 'author',
             'targetEntity' => $reviewerEntity,
-            'joinColumn' => [
+            'joinColumns' => [[
                 'name' => 'author_id',
                 'referencedColumnName' => $reviewerEntityMetadata->fieldMappings['id']['columnName'],
-                'nullable' => false,
+                'nullable' => true,
```

Comment:
Because of my previous comment. 

> In this method, the call to createProductReview is allowed to set the author of the review as null, which is not allowed by the mapping.

So, I've updated the mapping to follow the code expectation, but if a review should always have an author, then this should, indeed, be set to non-nullable and the code need to be updated to reflect that as well.